pkg/azureutils/azure_disk_utils.go

```diff
@@ -194,6 +185,44 @@ func GetCloudProvider(kubeconfig, secretName, secretNamespace, userAgent string)
 	return az, nil
 }
 
+// GetCloudProviderFromConfig get Azure Cloud Provider
+func GetCloudProvider(kubeConfig, secretName, secretNamespace, userAgent string) (*azure.Cloud, error) {
+	kubeClient, err := GetKubeClient(kubeConfig)
+	if err != nil && !os.IsNotExist(err) && err != rest.ErrNotInCluster {
+		return nil, fmt.Errorf("failed to get KubeClient: %v", err)
+	}
+	az, err := GetCloudProviderFromClient(kubeClient, secretName, secretNamespace, userAgent)
+	return az, err
+}
+
+// GetKubeConfig gets config object from config file
+func GetKubeConfig(kubeconfig string) (config *rest.Config, err error) {
+	// attempt to read the in-cluster config
+	config, err = rest.InClusterConfig()
```

Comment:
this func is changing the original behavior, any reason why change this?

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/36b2a9a078fb25728bb909045926e6abfa45565a/pkg/azureutils/azure_disk_utils.go#L445-L461

Reply:
I am trying to merge the behaviors we had in the v1 and v2 drivers. The recommended approach is to read the in cluster config and then fall back to kubeconfig if it fails. The behavior is essentially  the same in the old implementation as well. Was there a reason to prioritize kubeconfig over the in cluster one?

Reply:
If `kubeConfig` is not empty, it means that it was specified on the command line. In this case, it should probably take precedence here.

Reply:
But if we are running inside a cluster and our driver runs inside a pod, is there a scenario we where we would prefer the `kubeconfig` rather than the in-cluster config? From what I understand, the `kubeconfig` is mostly important for testing when the suite does not actually run inside a pod.

Reply:
if `kubeconfig` is set by user, it must be honored first:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/c1e583969e90b35140dea33d41ba5e2380b158ad/pkg/azurediskplugin/main.go#L44

Reply:
I have reverted the changes to the getter and kept the old logic.